test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Caching/CachingTestContext.cs

```diff
@@ -252,18 +252,18 @@ private void InitializeFiles()
             ProjectPath = Path.Combine(WorkingPath, "project.csproj");
 
             OutputPackagesPath = Path.Combine(WorkingPath, "packages");
-            CreateNuGetConfig(WorkingPath);
 
             Directory.CreateDirectory(OutputPackagesPath);
         }
 
-        private void CreateNuGetConfig(string workingDirectory)
+        internal void CreateNuGetConfig(string workingDirectory, string source)
         {
             string nugetConfigContent =
-                @"<?xml version='1.0' encoding='utf-8'?>
+                $@"<?xml version='1.0' encoding='utf-8'?>
 <configuration>
   <packageSources>
     <clear />
+    <add key=""http-feed"" value=""{source}"" allowInsecureConnections=""true""/>
```

Comment:
It looks like `allowInsecureConnections` setting can also be configured through an environment variable. I guess we need a test to verify that behavior as well. I am okay with addressing this comment in a follow up pull-request.

https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Configuration/Settings/Items/SourceItem.cs#L24-L36

Reply:
Are you referring to setting the config file as follows
```xml
<packageSources>
    <clear />    
    <add key="source" value="http://source" allowInsecureConnections="${Allow_Insecure_Connection_Value}" />
</packageSources>
```

Reply:
Yes, you are correct Nigusu. However, environment variable syntax is a bit different as I explained below. NuGet uses [Environment.ExpandEnvironment](https://learn.microsoft.com/dotnet/api/system.environment.expandenvironmentvariables) internally.

```xml
<packageSources>
    <clear />    
    <add key="source" value="http://source" allowInsecureConnections="%AllowInsecureConnectionValue%" />
</packageSources>
```

Reply:
Feel free to address this comment in a separate PR.

Reply:
I don't see any benefit with the environment variable for this setting either. I added the comment because currently code expands environment variable for this setting also. May be we could remove the environment variable support for this setting because it doesn't add much value to the users.